columnflow/plotting/plot_util.py

```diff
@@ -55,6 +57,97 @@ def get_cms_label(ax: plt.Axes, llabel: str) -> dict:
     return cms_label_kwargs
 
 
+def round_dynamic(value: int | float) -> int | float:
+    """
+    Rounds a *value* at various scales to a subjective, sensible precision. Rounding rules:
+
+        - 0 -> 0
+        - (0, 1) -> round to 0.1
+        - [1, 20]: round to 1
+        - (20, 100]: round to 5
+        - (100, 200]: round to 10
+        - (200, 500]: round to 20
+        - (500, 1000]: round to 50
+        - (1000, 2000]: round to 100
+        - (2000, 5000]: round to 200
+        - (5000, 10000]: round to 500
+        - ...
+
+    :param value: The value to round.
+    :return: The rounded value.
+    """
+    # separate sign
+    sign, value = (-1 if value < 0 else 1), abs(value)
+
+    # special cases
+    if not value:
+        # plain 0 int
+        return 0
+    if value < 1:
+        # round to float with 0.1 precision
+        return sign * round(value, 1)
+
+    # determine the reference scale
+    mag_over_1k = int(math.ceil(math.log10(value) - 3)) if value > 1000 else 0
+    value /= 10**mag_over_1k
+    if value <= 20:
+        ref = 1
+    elif value <= 100:
+        ref = 5
+    elif value <= 200:
+        ref = 10
+    elif value <= 500:
+        ref = 20
+    else:
+        ref = 50
+
+    # round and return
+    return sign * int(round(value / ref) * ref * 10**mag_over_1k)
+
+
+def inject_label(
```

Comment:
Can you provide some examples or tests for the specific usecases? I do not yet understand for what the `before_parentheses` mode is needed.

Reply:
In our signal samples, we usually have trailing parentheses to denote model parameters as in "HH_ggf (kl=1, kt=1)", and here one would naturally want to have the scaling factor before parameters as it could otherwise be interpreted as e.g. "parameters times 10". However, I just set the default to False to make this an opt-in behavior.